compiler/noirc_frontend/src/parser/parser/type_alias.rs

```diff
@@ -16,39 +19,49 @@ impl Parser<'_> {
     ) -> NoirTypeAlias {
         let Some(name) = self.eat_ident() else {
             self.expected_identifier();
-            return NoirTypeAlias {
+            return NoirTypeAlias::NormalTypeAlias(NormalTypeAlias {
                 visibility,
                 name: Ident::default(),
                 generics: Vec::new(),
                 typ: UnresolvedType { typ: UnresolvedTypeData::Error, location: Location::dummy() },
                 location: start_location,
-            };
+            });
         };
-
+        // Optional numeric type for alias over numeric generics
+        let mut num_typ = None;
         let generics = self.parse_generics();
-
-        if !self.eat_assign() {
+        if self.eat_colon() {
+            // To specify a type alias on a numeric generic expression, we need to specify the type of the expression
+            // It must be a numeric type
+            num_typ = Some(self.parse_type_or_error());
+        }
+        let location = self.location_since(start_location);
```

Comment:
This location should be created right before finishing parsing the type alias (I think this is why you get a difference in the tests). It seems previously this was right before the semicolons.

Reply:
I changed the location to be parsed before the semicolons, it fixed on unit test but not the ones in test/visibility.rs

Reply:
These unit tests are now working after an update from master.